src/eko/harmonics/__init__.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+"""
+This module contains some harmonics sum.
+Defintion are coming from :cite:`MuselliPhD,Bl_mlein_2000,Blumlein:2009ta`
+"""
+import numba as nb
+import numpy as np
+
+from .w1 import S1, Sm1
+from .w2 import S2, Sm2
+from .w3 import S3, S21, S2m1, Sm2m1, Sm3, Sm21
+from .w4 import S4, S31, S211, Sm4, Sm22, Sm31, Sm211
+from .w5 import S5, S23, S41, S221, S311, S2111, S2m3, S21m2, Sm5, Sm23, Sm221, Sm2111
+
+
+@nb.njit(cache=True)
+def base_harmonics_cache(n, max_weight=5, n_max_sums_weight=7):
```

Comment:
Hi @felixhekhorn, I'm trying to sort the harmonics in a more elegant way, but the thing is a bit involved:
* the best way would be to store them by weight.
* you also want the possibility to select the single index easily, since are used to compute more complex ones.
* the number of harmonics needed with a given weight depends also in the `PTO` and you don't want to compute unnecessary ones.

The solution I came up with is a matrix `(weight, max_n_harmonics_for_each_weight)` such that the first column always contains  the list of `S_{n}` and the last `S_{-n}` and the multi indices harmonics are stored in the middle always sorted by weight and index.  
For instance the complete cache will look: 
````
[
[S1,0,0,0,0,0,S-1],
[S2,0,0,0,0,0,S-2],
[S3,S21,S2-1,S-21,S-2-1,0,S-3],
[S4,S31,S211,S-22,S-211,S-31,S-4],
[S5,0,0,0,0,0,S-5],  # No need to store multi indices w5 harmonics 
]
````

It still looks a bit involved, maybe you have abetter idea?